sink/datadog.go

```diff
@@ -470,6 +504,41 @@ func (s *Datadog) estimateSize(metrics []datadogV2.MetricSeries) (int, error) {
 	return size / len(metrics), nil
 }
 
+func (s *Datadog) uniqIdentifierForMetric(name string, tags []string) string {
```

Comment:
This and `intForTag` aren't clear even though I have a rough idea of what you're trying to do: a metrics is unique only as `name+all tags`. And since some collectors get metrics for a long list of objects (all tables, for example), there can be a long list of unique metrics, like:
* count_read,db1,tbl1
* count_read,db1,tbl2
* ...
* count_read,dbN,tblN

What I don't get is how mappping to uint16 helps or works? (And why uint16 specifically, and not unit or uint32? And if "short" is the goal, why `+= "_"`?)

I don't think we need to sort because Go maps return hash values in consistent order.  Even even despite that, isn't a unique metric possible to compute by doing something like SHA(name,tags...) regardless of tag order?

Reply:
I original proposed a version of this. The idea is that metrics are distinct on name + tags/tag values, and for some metrics we would have a large variation. Large strings result in slower comparisons and additional memory overhead, so the idea was that instead of making a really big string and/or additionally hashing it that we map `tag:value` to an int sequentially as a very lightweight pseudo-hash in order to produce a unique key with a relatively smaller string. The `"_"` is unnecessary. There is also a bug in this implementation that I'll call out in a separate comment.

Reply:
> Even even despite that, isn't a unique metric possible to compute by doing something like SHA(name,tags...) regardless of tag order?

Thats a good point. I think a sha256 sum of (name, tags...) might just be enough. We can go with a sha sum and only consider optimising further if we see any performance issues.

> And if "short" is the goal, why += "_"?
I added that before when I was putting in whole tag values in the hash so in case I had to debug I would see them separated in the log. I can remove it.

Reply:
>Go maps return hash values in consistent order
I was so wrong: https://go.dev/ref/spec#For_statements and https://go.dev/doc/go1.3#map
I have a specific (but now faded) memory from long ago about this, but perhaps my memory got reversed over time: this changed (for small maps) way back in go1.3... or maybe I'm remember something else and the memory file has been corrupted to read as "map iteration."